apps/teams-test-app/src/components/NestedAppAuthAPIs.tsx

```diff
@@ -12,9 +30,144 @@ const NestedAppAuthAPIs = (): ReactElement => {
       onClick: async () => `NAA channel ${nestedAppAuth.isNAAChannelRecommended() ? 'is' : 'is not'} recommended`,
     });
 
+  const SendMessageToNestedAppAuthBridge = (): React.ReactElement =>
+    ApiWithTextInput({
+      name: 'SendMessageToNestedAppAuthBridge',
+      title: 'Send NAA Message to NestedAppAuth Bridge',
+      onClick: {
+        validateInput: () => {},
+        submit: async (input, setResult) => {
+          const bridge = (window as Window & { nestedAppAuthBridge?: NestedAppAuthBridge }).nestedAppAuthBridge;
+          if (!bridge) {
+            setResult('Bridge not available');
+            return 'Bridge not available';
+          }
+
+          // Define listener for responses
+          const listener = (response: unknown): void => {
+            setResult(JSON.stringify(response, null, 2));
+            bridge.removeEventListener?.('message', listener);
+          };
+
+          // Add event listener
+          bridge.addEventListener?.('message', listener);
+          bridge.postMessage?.(JSON.stringify(input));
+
+          setResult('Message sent successfully, awaiting response...');
+          return 'Message sent successfully';
+        },
+      },
+      defaultInput: NestedAppAuthRequest,
+    });
+
+  const SendMessageToTopWindow = (): React.ReactElement =>
+    ApiWithTextInput({
+      name: 'SendMessageToTopWindow',
+      title: 'Send NAA Message to Top Window',
```

Comment:
This says it is for sending NAA messages but it looks like it's really for sending any messages at all to the top window. Maybe a title change and/or functionality change so that the title matches the implementation?

Reply:
(This is basically the same feedback as above. There should be some validation that only NAA messages are used here probably)

Reply:
Thank you. Just added a validator. Please let me know in case of any further concern.